File: momentra/momentra/Shell/GroupEmpty/GroupLongFormShared.swift
import SwiftUI

let groupLocalOnlyNote = "Saved on this device — syncs when Group preferences API ships"

func experienceChipLabel(_ opt: GroupTypeOption) -> String {
    switch opt.code {
    case "TRIP": return "Trip"
    case "WEDDING": return "Wedding"
    case "HOUSE_PARTY": return "Party"
    case "OFFICE_OUTING": return "Office"
    default: return String(opt.label.prefix(8))
    }
}

func purchaseChipLabel(_ opt: GroupTypeOption) -> String {
    switch opt.code {
    case "GIFT_POOL": return "Gift"
    case "GROUP_PURCHASE": return "Purchase"
    case "SHARED_ASSET": return "Asset"
    case "COMMUNITY_PURCHASE", "CUSTOM": return "Custom"
    default: return String(opt.label.prefix(8))
    }
}

func livingChipLabel(_ opt: GroupTypeOption) -> String {
    switch opt.code {
    case "FLATMATES": return "Flatmates"
    case "FAMILY_HOUSEHOLD": return "Family"
    case "CO_LIVING": return "Co-living"
    case "COMMUNITY_LIVING", "CUSTOM": return "Custom"
    default: return String(opt.label.prefix(8))
    }
}

struct GroupLongFormTypeChipStrip: View {
    let title: String
    let types: [GroupTypeOption]
    let selectedCode: String
    var shortLabel: (GroupTypeOption) -> String
    var onSelect: (GroupTypeOption) -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(title.uppercased())
                .font(.plusJakarta(size: 9, weight: .medium))
                .foregroundStyle(Color(hex: "#938EA1"))
                .tracking(0.7)
                .padding(.horizontal, 4)
            HStack(spacing: 6) {
                ForEach(types) { opt in
                    let selected = opt.code == selectedCode
                    let accent = GroupSetupTheme.palette(for: opt.code).accent
                    Button { onSelect(opt) } label: {
                        VStack(spacing: 4) {
                            Image(opt.iconName ?? "ges_type_trip")
                                .renderingMode(.template)
                                .resizable()
                                .scaledToFit()
                                .frame(width: 18, height: 18)
                                .foregroundStyle(selected ? accent : GroupSetupTheme.textSecondary)
                            Text(shortLabel(opt))
                                .font(.plusJakarta(size: 8.5, weight: .medium))
                                .foregroundStyle(selected ? accent : GroupSetupTheme.textSecondary)
                                .lineLimit(1)
                        }
                        .frame(maxWidth: .infinity, minHeight: 56)
                        .background(
                            (selected ? accent.opacity(0.16) : Color(hex: "#1E293B")),
                            in: RoundedRectangle(cornerRadius: 14)
                        )
                        .overlay(
                            RoundedRectangle(cornerRadius: 14)
                                .stroke(selected ? accent.opacity(0.6) : Color(hex: "#3A3842").opacity(0.8), lineWidth: 1)
                        )
                    }
                    .buttonStyle(.plain)
                }
            }
        }
        .padding(8)
        .background(Color(hex: "#161B26"), in: RoundedRectangle(cornerRadius: 18))
        .overlay(RoundedRectangle(cornerRadius: 18).stroke(Color(hex: "#3A3842").opacity(0.75), lineWidth: 1))
    }
}

struct GroupLongFormDiamondDivider: View {
    var body: some View {
        HStack(spacing: 12) {
            Rectangle().fill(Color(hex: "#3A3842")).frame(height: 1)
            RoundedRectangle(cornerRadius: 1).fill(Color(hex: "#C9BFFF")).frame(width: 8, height: 8)
            Rectangle().fill(Color(hex: "#3A3842")).frame(height: 1)
        }
        .padding(.vertical, 16)
    }
}

struct GroupLongFormSectionCard<Content: View>: View {
    let step: String
    let title: String
    let accent: Color
    @ViewBuilder var content: () -> Content

    var body: some View {
        VStack(alignment: .leading, spacing: 20) {
            HStack(spacing: 12) {
                Text(step)
                    .font(.plusJakarta(size: 64, weight: .heavy))
                    .foregroundStyle(accent.opacity(0.06))
                Text(title.uppercased())
                    .font(.plusJakarta(size: 11, weight: .semibold))
                    .foregroundStyle(accent)
            }
            content()
        }
        .padding(24)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color(hex: "#3A3842"), in: RoundedRectangle(cornerRadius: 16))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(Color(hex: "#938EA1").opacity(0.35), lineWidth: 1))
    }
}

let groupDestinationSuggestions = ["Goa, India", "Jaipur", "Manali", "Singapore"]

struct GroupLongFormDestinationField: View {
    let label: String
    let hint: String
    @Binding var value: String
    var placeholder: String = "City, country or venue"
    var suggestions: [String] = groupDestinationSuggestions
    var testTag: String = "setup.field.destination"

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(label)
                .font(.plusJakarta(size: 14, weight: .medium))
                .foregroundStyle(GroupSetupTheme.textSecondary)
            Text(hint)
                .font(.plusJakarta(size: 12))
                .foregroundStyle(GroupSetupTheme.textSecondary.opacity(0.85))
            TextField(placeholder, text: $value)
                .font(.plusJakarta(size: 16, weight: .semibold))
                .foregroundStyle(Color(hex: "#E5E0EE"))
                .padding(.horizontal, 12)
                .padding(.vertical, 10)
                .background(Color(hex: "#161B26"), in: RoundedRectangle(cornerRadius: 8))
                .overlay(RoundedRectangle(cornerRadius: 8).stroke(Color(hex: "#1E293B"), lineWidth: 1))
                .accessibilityIdentifier(testTag)
                .onChange(of: value) { _, newValue in
                    if newValue.count > 200 {
                        value = String(newValue.prefix(200))
                    }
                }
            if !suggestions.isEmpty {
                FlowLayout(spacing: 8) {
                    ForEach(suggestions, id: \.self) { suggestion in
                        let selected = value.caseInsensitiveCompare(suggestion) == .orderedSame
                        Button {
                            value = suggestion
                        } label: {
                            Text(suggestion)
                                .font(.plusJakarta(size: 12, weight: selected ? .semibold : .regular))
                                .foregroundStyle(selected ? Color(hex: "#C9BFFF") : GroupSetupTheme.textSecondary)
                                .padding(.horizontal, 12)
                                .padding(.vertical, 6)
                                .background(
                                    (selected ? Color(hex: "#2D2640") : Color(hex: "#161B26")),
                                    in: Capsule()
                                )
                                .overlay(
                                    Capsule().stroke(selected ? Color(hex: "#7C5CFC") : Color(hex: "#1E293B"), lineWidth: 1)
                                )
                        }
                        .buttonStyle(.plain)
                    }
                }
            }
        }
        .padding(.vertical, 8)
    }
}

struct GroupBudgetCustomField: View {
    @Binding var value: String
    let currencyCode: String
    var testTag: String = "setup.field.budgetCustom"

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text("Custom amount (\(currencyCode))")
                .font(.plusJakarta(size: 12))
                .foregroundStyle(GroupSetupTheme.textSecondary)
            TextField("84,000", text: $value)
                .font(.plusJakarta(size: 16, weight: .semibold))
                .foregroundStyle(Color(hex: "#E5E0EE"))
                .keyboardType(.numberPad)
                .padding(.horizontal, 12)
                .padding(.vertical, 10)
                .background(Color(hex: "#161B26"))
                .clipShape(RoundedRectangle(cornerRadius: 8))
                .overlay(RoundedRectangle(cornerRadius: 8).stroke(Color(hex: "#1E293B")))
                .accessibilityIdentifier(testTag)
                .onChange(of: value) { _, newValue in
                    let formatted = GroupBudgetUtils.formatCustomAmountInput(newValue)
                    if formatted != newValue {
                        value = formatted
                    }
                }
        }
    }
}

struct GroupLongFormPrefRow: View {
    let label: String
    let hint: String
    let value: String
    let options: [String]
    let onValueChange: (String) -> Void
    var editableGlyph: Bool = false
    var testTag: String = "setup.dropdown"

    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 4) {
                Text(label).font(.plusJakarta(size: 14, weight: .medium)).foregroundStyle(GroupSetupTheme.textSecondary)
                Text(hint).font(.plusJakarta(size: 12)).foregroundStyle(GroupSetupTheme.textSecondary.opacity(0.85))
            }
            Spacer(minLength: 8)
            Menu {
                ForEach(options, id: \.self) { option in
                    Button(option) { onValueChange(option) }
                }
            } label: {
                HStack(spacing: 8) {
                    Text(value).font(.plusJakarta(size: 13, weight: .semibold)).foregroundStyle(Color(hex: "#E5E0EE")).lineLimit(1)
                    Text(editableGlyph ? "✎" : "▼").font(.plusJakarta(size: 10, weight: .semibold)).foregroundStyle(GroupSetupTheme.textSecondary)
                }
                .padding(.horizontal, 12).padding(.vertical, 6)
                .background(Color(hex: "#161B26"), in: RoundedRectangle(cornerRadius: 8))
                .overlay(RoundedRectangle(cornerRadius: 8).stroke(Color(hex: "#1E293B"), lineWidth: 1))
            }
            .accessibilityIdentifier("\(testTag).trigger")
        }
        .padding(.vertical, 14)
        .accessibilityIdentifier(testTag)
    }
}

struct GroupLongFormToggleRow: View {
    let title: String
    let subtitle: String
    @Binding var checked: Bool
    let accent: Color

    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 4) {
                Text(title).font(.plusJakarta(size: 14, weight: .medium)).foregroundStyle(GroupSetupTheme.textSecondary)
                Text(subtitle).font(.plusJakarta(size: 12)).foregroundStyle(GroupSetupTheme.textSecondary.opacity(0.85))
            }
            Spacer()
            Toggle("", isOn: $checked).labelsHidden().tint(accent)
        }
        .padding(.vertical, 14)
    }
}

struct GroupLongFormNamePills: View {
    let primary: String
    let secondary: String
    let accent: Color

    var body: some View {
        HStack(spacing: 8) {
            ForEach([primary, secondary], id: \.self) { text in
                HStack(spacing: 8) {
                    Circle().fill(accent).frame(width: 6, height: 6)
                    Text(text).font(.plusJakarta(size: 14, weight: .medium)).foregroundStyle(Color(hex: "#E5E0EE")).lineLimit(1)
                    Text("▼").font(.plusJakarta(size: 12, weight: .semibold)).foregroundStyle(GroupSetupTheme.textSecondary)
                }
                .padding(.horizontal, 16).padding(.vertical, 10)
                .background(Color(hex: "#161B26"), in: Capsule())
                .overlay(Capsule().stroke(Color(hex: "#1E293B"), lineWidth: 1))
            }
        }
    }
}

struct GroupLongFormLocalOnlyNote: View {
    var body: some View {
        Text(groupLocalOnlyNote)
            .font(.plusJakarta(size: 11))
            .foregroundStyle(GroupSetupTheme.textSecondary)
            .frame(maxWidth: .infinity, alignment: .leading)
    }
}

struct GroupLongFormReadyBanner: View {
    let message: String

    var body: some View {
        HStack(spacing: 10) {
            Text("✓").font(.plusJakarta(size: 14, weight: .bold)).foregroundStyle(Color(hex: "#10B981"))
            Text(message).font(.plusJakarta(size: 13, weight: .semibold)).foregroundStyle(Color(hex: "#E5E0EE"))
        }
        .padding(.horizontal, 16).padding(.vertical, 12)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Color(hex: "#10B981").opacity(0.15), in: Capsule())
        .overlay(Capsule().stroke(Color(hex: "#10B981").opacity(0.35), lineWidth: 1))
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/empty/group/GroupLongFormShared.kt
package com.example.momentra.ui.shell.empty.group

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.ColorFilter
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import com.example.momentra.ui.shell.maestro.MaestroIds
import androidx.compose.ui.platform.testTag
import com.example.momentra.ui.theme.PlusJakartaSans

internal const val GROUP_LOCAL_ONLY_NOTE =
    "Synced with your group — budget and reminders save on activate"

/** Compact Figma type strip (EXPERIENCE SETUPS / PURCHASE SETUPS / …). */
@Composable
fun GroupLongFormTypeChipStrip(
    title: String,
    types: List<GroupTypeOption>,
    selectedCode: String,
    onSelect: (GroupTypeOption) -> Unit,
    shortLabel: (GroupTypeOption) -> String,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(18.dp))
            .background(Color(0xFF161B26))
            .border(1.dp, Color(0xBF3A3842), RoundedCornerShape(18.dp))
            .padding(horizontal = 8.dp, vertical = 8.dp),
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Text(
            title.uppercase(),
            color = Color(0xFF938EA1),
            fontSize = 9.sp,
            fontWeight = FontWeight.Medium,
            fontFamily = PlusJakartaSans,
            letterSpacing = 0.7.sp,
            modifier = Modifier.padding(horizontal = 4.dp),
        )
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(6.dp),
        ) {
            types.forEach { opt ->
                val selected = opt.code == selectedCode
                val accent = opt.palette.accent
                Column(
                    modifier = Modifier
                        .weight(1f)
                        .height(56.dp)
                        .clip(RoundedCornerShape(14.dp))
                        .background(if (selected) accent.copy(alpha = 0.16f) else Color(0xFF1E293B))
                        .border(
                            1.dp,
                            if (selected) accent.copy(alpha = 0.6f) else Color(0xCC3A3842),
                            RoundedCornerShape(14.dp),
                        )
                        .clickable { onSelect(opt) }
                        .padding(vertical = 5.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(4.dp),
                ) {
                    Image(
                        painterResource(opt.iconRes),
                        contentDescription = null,
                        modifier = Modifier.size(18.dp),
                        colorFilter = ColorFilter.tint(if (selected) accent else GroupSetupTheme.TextSecondary),
                    )
                    Text(
                        shortLabel(opt),
                        color = if (selected) accent else GroupSetupTheme.TextSecondary,
                        fontSize = 8.5.sp,
                        fontWeight = FontWeight.Medium,
                        fontFamily = PlusJakartaSans,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis,
                        textAlign = TextAlign.Center,
                    )
                }
            }
        }
    }
}

@Composable
fun GroupLongFormDiamondDivider(modifier: Modifier = Modifier) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        HorizontalDivider(modifier = Modifier.weight(1f), color = Color(0xFF3A3842), thickness = 1.dp)
        Box(
            modifier = Modifier
                .size(8.dp)
                .clip(RoundedCornerShape(1.dp))
                .background(Color(0xFFC9BFFF)),
        )
        HorizontalDivider(modifier = Modifier.weight(1f), color = Color(0xFF3A3842), thickness = 1.dp)
    }
}

@Composable
fun GroupLongFormSectionCard(
    step: String,
    title: String,
    accent: Color,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(Color(0xFF3A3842))
            .border(1.dp, Color(0xFF938EA1).copy(alpha = 0.35f), RoundedCornerShape(16.dp))
            .padding(24.dp),
        verticalArrangement = Arrangement.spacedBy(20.dp),
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            Box {
                Text(
                    step,
                    color = accent.copy(alpha = 0.06f),
                    fontSize = 64.sp,
                    fontWeight = FontWeight.ExtraBold,
                    fontFamily = PlusJakartaSans,
                )
            }
            Text(
                title.uppercase(),
                color = accent,
                fontSize = 11.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = PlusJakartaSans,
            )
        }
        content()
    }
}

@Composable
fun GroupLongFormSubsectionTitle(text: String, modifier: Modifier = Modifier) {
    Text(
        text.uppercase(),
        color = GroupSetupTheme.TextSecondary,
        fontSize = 10.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = PlusJakartaSans,
        modifier = modifier.fillMaxWidth(),
    )
}

@Composable
fun GroupLongFormGroupTitle(text: String, modifier: Modifier = Modifier) {
    Text(
        text,
        color = Color(0xFFE5E0EE),
        fontSize = 15.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = PlusJakartaSans,
        modifier = modifier.fillMaxWidth(),
    )
}

/** Quick picks for shared-experience destination — user can also type any custom value. */
val groupDestinationSuggestions = listOf("Goa, India", "Jaipur", "Manali", "Singapore")

/** Label + hint + free-text destination with optional suggestion chips. */
@OptIn(ExperimentalLayoutApi::class)
@Composable
fun GroupLongFormDestinationField(
    label: String,
    hint: String,
    value: String,
    onValueChange: (String) -> Unit,
    placeholder: String = "City, country or venue",
    suggestions: List<String> = groupDestinationSuggestions,
    modifier: Modifier = Modifier,
    testTag: String = MaestroIds.setupField("destination"),
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Text(
            label,
            color = GroupSetupTheme.TextSecondary,
            fontSize = 14.sp,
            fontWeight = FontWeight.Medium,
            fontFamily = PlusJakartaSans,
        )
        Text(
            hint,
            color = GroupSetupTheme.TextSecondary.copy(alpha = 0.85f),
            fontSize = 12.sp,
            fontFamily = PlusJakartaSans,
        )
        androidx.compose.foundation.text.BasicTextField(
            value = value,
            onValueChange = { if (it.length <= 200) onValueChange(it) },
            singleLine = true,
            textStyle = androidx.compose.ui.text.TextStyle(
                color = Color(0xFFE5E0EE),
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = PlusJakartaSans,
            ),
            cursorBrush = androidx.compose.ui.graphics.SolidColor(Color(0xFFC9BFFF)),
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(8.dp))
                .background(Color(0xFF161B26))
                .border(1.dp, Color(0xFF1E293B), RoundedCornerShape(8.dp))
                .padding(horizontal = 12.dp, vertical = 10.dp)
                .testTag(testTag),
            decorationBox = { inner ->
                if (value.isEmpty()) {
                    Text(
                        placeholder,
                        color = GroupSetupTheme.TextSecondary.copy(alpha = 0.7f),
                        fontSize = 16.sp,
                        fontFamily = PlusJakartaSans,
                    )
                }
                inner()
            },
        )
        if (suggestions.isNotEmpty()) {
            FlowRow(
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                suggestions.forEach { suggestion ->
                    val selected = value.equals(suggestion, ignoreCase = true)
                    Text(
                        suggestion,
                        color = if (selected) Color(0xFFC9BFFF) else GroupSetupTheme.TextSecondary,
                        fontSize = 12.sp,
                        fontWeight = if (selected) FontWeight.SemiBold else FontWeight.Normal,
                        fontFamily = PlusJakartaSans,
                        modifier = Modifier
                            .clip(RoundedCornerShape(999.dp))
                            .background(
                                if (selected) Color(0xFF2D2640) else Color(0xFF161B26),
                            )
                            .border(
                                1.dp,
                                if (selected) Color(0xFF7C5CFC) else Color(0xFF1E293B),
                                RoundedCornerShape(999.dp),
                            )
                            .clickable { onValueChange(suggestion) }
                            .padding(horizontal = 12.dp, vertical = 6.dp)
                            .testTag("$testTag.suggestion.${suggestion.hashCode()}"),
                    )
                }
            }
        }
    }
}

/** Label + hint + trailing dropdown pill (Figma form-row). */
@Composable
fun GroupLongFormPrefRow(
    label: String,
    hint: String,
    value: String,
    options: List<String>,
    onValueChange: (String) -> Unit,
    modifier: Modifier = Modifier,
    editableGlyph: Boolean = false,
    testTag: String = MaestroIds.SETUP_DROPDOWN,
) {
    var expanded by remember { mutableStateOf(false) }
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween,
    ) {
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(4.dp),
        ) {
            Text(
                label,
                color = GroupSetupTheme.TextSecondary,
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
                fontFamily = PlusJakartaSans,
            )
            Text(
                hint,
                color = GroupSetupTheme.TextSecondary.copy(alpha = 0.85f),
                fontSize = 12.sp,
                fontFamily = PlusJakartaSans,
            )
        }
        Spacer(Modifier.width(8.dp))
        Box {
            Row(
                modifier = Modifier
                    .clip(RoundedCornerShape(8.dp))
                    .background(Color(0xFF161B26))
                    .border(1.dp, Color(0xFF1E293B), RoundedCornerShape(8.dp))
                    .clickable { expanded = true }
                    .padding(horizontal = 12.dp, vertical = 6.dp)
                    .testTag("$testTag.trigger"),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                Text(
                    value,
                    color = Color(0xFFE5E0EE),
                    fontSize = 13.sp,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = PlusJakartaSans,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
                Text(
                    if (editableGlyph) "✎" else "▼",
                    color = GroupSetupTheme.TextSecondary,
                    fontSize = 10.sp,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = PlusJakartaSans,
                )
            }
            DropdownMenu(
                expanded = expanded,
                onDismissRequest = { expanded = false },
                modifier = Modifier.testTag("$testTag.menu"),
            ) {
                options.forEach { option ->
                    DropdownMenuItem(
                        text = {
                            Text(
                                option,
                                color = if (option == value) Color(0xFFC9BFFF) else Color.White,
                                fontWeight = if (option == value) FontWeight.Bold else FontWeight.Normal,
                                fontFamily = PlusJakartaSans,
                            )
                        },
                        onClick = {
                            onValueChange(option)
                            expanded = false
                        },
                    )
                }
            }
        }
    }
}

/** Custom budget amount field shown when Budget preset is "Custom…". */
@Composable
fun GroupBudgetCustomField(
    value: String,
    onValueChange: (String) -> Unit,
    currencyCode: String,
    modifier: Modifier = Modifier,
    testTag: String = MaestroIds.setupField("budgetCustom"),
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(bottom = 8.dp),
        verticalArrangement = Arrangement.spacedBy(6.dp),
    ) {
        Text(
            "Custom amount ($currencyCode)",
            color = GroupSetupTheme.TextSecondary,
            fontSize = 12.sp,
            fontFamily = PlusJakartaSans,
        )
        androidx.compose.foundation.text.BasicTextField(
            value = value,
            onValueChange = { onValueChange(GroupBudgetUtils.formatCustomAmountInput(it)) },
            singleLine = true,
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
            textStyle = androidx.compose.ui.text.TextStyle(
                color = Color(0xFFE5E0EE),
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = PlusJakartaSans,
            ),
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(8.dp))
                .background(Color(0xFF161B26))
                .border(1.dp, Color(0xFF1E293B), RoundedCornerShape(8.dp))
                .padding(horizontal = 12.dp, vertical = 10.dp)
                .testTag(testTag),
            decorationBox = { inner ->
                if (value.isEmpty()) {
                    Text(
                        "₹84,000",
                        color = GroupSetupTheme.TextSecondary.copy(alpha = 0.6f),
                        fontSize = 16.sp,
                        fontFamily = PlusJakartaSans,
                    )
                }
                inner()
            },
        )
    }
}

@Composable
fun GroupLongFormToggleRow(
    title: String,
    subtitle: String,
    checked: Boolean,
    onCheckedChange: (Boolean) -> Unit,
    accent: Color,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween,
    ) {
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(4.dp),
        ) {
            Text(
                title,
                color = GroupSetupTheme.TextSecondary,
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
                fontFamily = PlusJakartaSans,
            )
            Text(
                subtitle,
                color = GroupSetupTheme.TextSecondary.copy(alpha = 0.85f),
                fontSize = 12.sp,
                fontFamily = PlusJakartaSans,
            )
        }
        Switch(
            checked = checked,
            onCheckedChange = onCheckedChange,
            colors = SwitchDefaults.colors(
                checkedThumbColor = Color.White,
                checkedTrackColor = accent,
                uncheckedThumbColor = GroupSetupTheme.TextSecondary,
                uncheckedTrackColor = GroupSetupTheme.Border,
            ),
        )
    }
}

@Composable
fun GroupLongFormNamePills(
    primary: String,
    secondary: String,
    accent: Color,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        listOf(primary to true, secondary to false).forEach { (text, _) ->
            Row(
                modifier = Modifier
                    .clip(RoundedCornerShape(20.dp))
                    .background(Color(0xFF161B26))
                    .border(1.dp, Color(0xFF1E293B), RoundedCornerShape(20.dp))
                    .padding(horizontal = 16.dp, vertical = 10.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                Box(
                    modifier = Modifier
                        .size(6.dp)
                        .clip(RoundedCornerShape(3.dp))
                        .background(accent),
                )
                Text(
                    text,
                    color = Color(0xFFE5E0EE),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    fontFamily = PlusJakartaSans,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
                Text("▼", color = GroupSetupTheme.TextSecondary, fontSize = 12.sp, fontWeight = FontWeight.SemiBold)
            }
        }
    }
}

@Composable
fun GroupLongFormLocalOnlyNote(modifier: Modifier = Modifier) {
    Text(
        GROUP_LOCAL_ONLY_NOTE,
        color = GroupSetupTheme.TextSecondary,
        fontSize = 11.sp,
        fontFamily = PlusJakartaSans,
        modifier = modifier.fillMaxWidth(),
    )
}

@Composable
fun GroupLongFormReadyBanner(
    message: String,
    accent: Color,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(999.dp))
            .background(Color(0xFF10B981).copy(alpha = 0.15f))
            .border(1.dp, Color(0xFF10B981).copy(alpha = 0.35f), RoundedCornerShape(999.dp))
            .padding(horizontal = 16.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Text("✓", color = Color(0xFF10B981), fontSize = 14.sp, fontWeight = FontWeight.Bold)
        Text(
            message,
            color = Color(0xFFE5E0EE),
            fontSize = 13.sp,
            fontWeight = FontWeight.SemiBold,
            fontFamily = PlusJakartaSans,
        )
    }
}

internal fun cycleOption(current: String, options: List<String>): String {
    if (options.isEmpty()) return current
    val idx = options.indexOf(current).let { if (it < 0) 0 else it }
    return options[(idx + 1) % options.size]
}

internal fun experienceChipLabel(opt: GroupTypeOption): String = when (opt.code) {
    "TRIP" -> "Trip"
    "WEDDING" -> "Wedding"
    "HOUSE_PARTY" -> "Celebration"
    "OFFICE_OUTING" -> "Office"
    else -> opt.label.take(8)
}

internal fun purchaseChipLabel(opt: GroupTypeOption): String = when (opt.code) {
    "GIFT_POOL" -> "Gift"
    "GROUP_PURCHASE" -> "Purchase"
    "SHARED_ASSET" -> "Asset"
    "COMMUNITY_PURCHASE", "CUSTOM" -> "Custom"
    else -> opt.label.take(8)
}

internal fun livingChipLabel(opt: GroupTypeOption): String = when (opt.code) {
    "FLATMATES" -> "Flatmates"
    "FAMILY_HOUSEHOLD" -> "Family"
    "CO_LIVING" -> "Co-living"
    "CUSTOM" -> "Custom"
    else -> opt.label.take(8)
}
